### PR TITLE
Fredhase/codeclimate refactor onboarding initializers 3739

### PR DESCRIPTION
**Pull request by fredhase**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Refactoring codes in app/assets/javascripts/initializers (DrawerSliders and SwipeGestures). Some minor refactoring and some repetitive blocks are now a function.
In app/javascript/onboarding/components just made minor refactoring.

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

### CODE REVIEW

File: app/assets/javascripts/initializers/initializeDrawerSliders.js [hunk 1]
```diff
@@ -1,57 +1,49 @@
...
       }
-    }
-    if (document.getElementById("on-page-nav-butt-right")){
-      document.getElementById("on-page-nav-butt-right").onclick = function(){
-        swipeState = "right"
-        slideSidebar("right","intoView");
-      }
-    }
-    InstantClick.on('change', function() {
-      document.getElementsByTagName('body')[0].classList.remove('modal-open');
```
Comment (paulodiovani) [inline app/assets/javascripts/initializers/initializeDrawerSliders.js]: This `classList.remove('modal-open')` seems to be missed in the new code.

File: app/assets/javascripts/initializers/initializeSwipeGestures.js [hunk 2]
```diff
@@ -1,61 +1,97 @@
...
-  }
-  else{
-    swipeState = "middle"
-    slideSidebar("left","outOfView");
+  if (swipeState == 'middle') {
+    swipeState = 'right';
+    slideSidebar('right', 'intoView');
+  } else {
+    swipeState = 'middle';
+    slideSidebar('left', 'outOfView');
```
Comment (paulodiovani) [inline app/assets/javascripts/initializers/initializeSwipeGestures.js]: you can avoid using `else` at all

```js

  if (!document.getElementById('on-page-nav-controls')) {
    return;
  }

  if (swipeState == 'middle') {
    swipeState = 'right';
    return slideSidebar('right', 'intoView');
  }

  swipeState = 'middle';
  slideSidebar('left', 'outOfView');

File: app/javascript/onboarding/components/ClosingSlide.jsx [hunk 3]
```diff
@@ -125,42 +138,10 @@ class ClosingSlide extends Component {
           </div>
         );
       }
-      return (
+      return(
```
Comment (paulodiovani) [inline app/javascript/onboarding/components/ClosingSlide.jsx]: Missing a blank space

File: app/javascript/onboarding/components/ClosingSlide.jsx [hunk 4]
```diff
@@ -1,22 +1,35 @@
...
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify({ user: { last_onboarding_page: 'closing slide' } }),
-      credentials: 'same-origin',
-    });
+    updateOnboarding('closing slide');
   }
 
+  whatsNext() {
+    const properties = [
```
Comment (paulodiovani) [inline app/javascript/onboarding/components/ClosingSlide.jsx]: please, rename this to something like `whatsNextLinks`,

or just return the array (see next comment)

File: app/javascript/onboarding/components/ClosingSlide.jsx [hunk 5]
```diff
@@ -1,22 +1,35 @@
...
+          <a key={ index } href={ element.href } data-no-instant={(index === 0) ? true : null }>
+            { element.text }
+            <p className="whatnext-emoji">
+              <span role="img" aria-label="tada">
+                { element.emoji }
+              </span>
+            </p>
+          </a>   
+      )
+    );
```
Comment (paulodiovani) [inline app/javascript/onboarding/components/ClosingSlide.jsx]: Doesn't add much to have a method return a small piece of component here.
As a suggestion:

1. make `whatsNext()` method return just the array of objects.
2. `.map` it in `render()` method

**Comment by fredhase** [comment]
@paulodiovani 
Thanks for the review and tips. Unfortunattely, I'm still studying unit tests, so, for now, I can't do much about that. But I fixed everything pointed out :)

**Comment by fredhase** [comment]
@nickytonline 
Hi, sorry. I'm still learning about PRs and branches and I may have forgotten to make the checkout to the master before starting a new branch. So, i'm not sure how my branchs are right now :thinking: 
Really sorry about the mess

**Comment by nickytonline** [comment]
> @nickytonline
> Hi, sorry. I'm still learning about PRs and branches and I may have forgotten to make the checkout to the master before starting a new branch. So, i'm not sure how my branchs are right now 🤔
> Really sorry about the mess

No worries @fredhase. And no pressure no get this done. If you need some assistance with branches and git, happy to help. This post from one of our community members may be of help to you, https://dev.to/unseenwizzard/learn-git-concepts-not-commands-4gjc

**Comment by fredhase** [comment]
Closing this PR for some branche mistakes that caused conflict

**Pull request closed without merging**